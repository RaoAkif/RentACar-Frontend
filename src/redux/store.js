import { configureStore } from '@reduxjs/toolkit';
import cakeReducer from './cake/cakeSlice';
import icecreamReducer from './icecream/icecreamSlice';
import userReducer from './user/userSlice';

const store = configureStore({
  reducer: {
    cake: cakeReducer,
    icecream: icecreamReducer,
    user: userReducer,
  },
});

export default store;
