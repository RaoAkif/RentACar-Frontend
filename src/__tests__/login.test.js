import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import { render } from '@testing-library/react';
import Login from '../pages/auth/login';

describe('Login', () => {  
  it('should render correctly', () => {
    const { container } = render(
      <BrowserRouter>
        <Provider store={store}>
          <Login />
        </Provider>,
      </BrowserRouter>,
    );

    expect(container).toMatchSnapshot();
  });
});
