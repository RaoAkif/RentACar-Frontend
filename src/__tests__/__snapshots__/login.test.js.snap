// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should render correctly 1`] = `
<div>
  <div
    class="auth-page"
  >
    <div
      class="container"
    >
      <img
        alt="logo"
        class="form-logo"
        src="rentacar_logo.png"
      />
      <form
        action=""
        class="form"
        method="POST"
      >
        <div
          class="form-group"
        >
          <input
            class="form-control"
            id="name"
            placeholder="Name..."
            type="text"
            value=""
          />
        </div>
        <button
          class="btn btn-green"
          type="submit"
        >
          Login
        </button>
      </form>
      <div
        class="navi-option"
      >
        <p>
          Don't have an account?
          <a
            href="/signup"
          >
             Signup
          </a>
        </p>
      </div>
    </div>
  </div>
  ,
</div>
`;
